packages/components/src/InputDate/InputDate.rebuilt.tsx

```diff
@@ -61,8 +59,21 @@ export const InputDateRebuilt = forwardRef(function InputDateInternal(
       // We prevent the picker from opening on focus for keyboard navigation, so to maintain a good UX for mouse users we want to open the picker on click
       <div onClick={onClick}>
         <InputText
-          {...newActivatorProps}
-          {...props}
```

Comment:
I'm going to write a hefty ticket about these two sets of props. there's a ton to unpack that I won't go into for the purposes of this bugfix.

the shortest possible version of it is that spreading like this is hiding a lot. there are numerous invalid props, and overlapping props that will be stomped on by the second spread. Typescript isn't able to do its job when we spread

<img width="362" alt="image" src="https://github.com/user-attachments/assets/ccf061f6-c0e3-4a4b-ad63-eec69c0653df" />